Mathlib/Algebra/Homology/ShortComplex/ExactFunctor.lean

```diff
@@ -46,6 +47,173 @@ noncomputable def preservesFiniteColimitsOfPreservesHomology
     ⟨F.obj 0, by rw [IsZero.iff_id_eq_zero, ← F.map_id, id_zero, F.map_zero]⟩
   exact preservesFiniteColimitsOfPreservesCokernels F
 
+end
+
+
+section
+
+variable {C D : Type*} [Category C] [Category D] [Abelian C] [Abelian D]
+variable (F : C ⥤ D) [F.Additive]
+
+/--
+If a functor `F : C ⥤ D` preserves exact sequences, then it preserves monomorphism.
+-/
+def preservesMonomorphismsOfPreservesShortComplexExact
+    (h : ∀ (S : ShortComplex C), S.Exact → (S.map F).Exact) :
+    F.PreservesMonomorphisms where
+  preserves {X Y} f m :=
+    ShortComplex.exact_iff_mono (hf := by simp) |>.1 $ h _ $
+      ShortComplex.exact_iff_mono (.mk (0 : 0 ⟶ X) f (by simp)) rfl |>.2 m
+
+/--
+If a functor `F : C ⥤ D` preserves exact sequences, then it preserves epimorphism.
+-/
+def preservesEpimorphismsOfPreservesShortComplexExact
+    (h : ∀ (S : ShortComplex C), S.Exact → (S.map F).Exact) :
+    F.PreservesEpimorphisms where
+  preserves {X Y} f e :=
+    ShortComplex.exact_iff_epi (hg := by simp) |>.1 $ h _ $
+      ShortComplex.exact_iff_epi (.mk f (0 : Y ⟶ 0) (by simp)) rfl |>.2 e
+
+/--
+If a functor `F : C ⥤ D` preserves exact sequences, then it preserves short exact sequences.
+-/
+lemma preserves_shortComplex_shortExact_of_preserves_shortComplex_exact
+    (h : ∀ (S : ShortComplex C), S.Exact → (S.map F).Exact)
+    (S : ShortComplex C) (hS : S.ShortExact) : (S.map F).ShortExact where
+  exact := h _ hS.exact
+  mono_f :=
+    letI := F.preservesMonomorphismsOfPreservesShortComplexExact h
+    letI : Mono S.f := hS.mono_f
+    map_mono _ _
+  epi_g :=
+    letI := F.preservesEpimorphismsOfPreservesShortComplexExact h
+    letI : Epi S.g := hS.epi_g
+    map_epi _ _
+
+set_option maxHeartbeats 500000 in
+/--
+If a functor `F : C ⥤ D` preserves exact sequences, then it preserves kernels.
+-/
+noncomputable def preservesKernelsOfPreservesShortComplexExact
```

Comment:
There is probably some range of improvements in this definition, but more generally, I would suggest proving a TFAE for an additive functor between abelian categories:

1. If `S` is *short* exact, then `S.map F` is exact and `F.map S.f` is mono;
2. If `S` is exact and `S.f` is mono, the same applies to `S.map F`.
3. `F` preserves kernels.
4. `F` preserves finite limits.

Using the dual result, we should get that `F` preserves `ShortComplex.ShortExact` iff it preserves both finite limits and finite colimits, and with some more work we would also obtain that this is equivalent to preserving `ShortComplex.Exact` (and preserving homology). (You have obtained this, but it is also an opportunity to formalise criteria for left exact and right exact functors.)

Reply:
As I am removing the old homology API (which implies some refactor work in PR #14569), it turns out that we essentially already have the equivalences you should need for the exactness of tensor product (see files `CategoryTheory.Abelian.Exact`, `Algebra.Homology.ShortComplex.PreservesHomology` and `Algebra.Homology.ShortComplex.ExactFunctor`).

Reply:
> There is probably some range of improvements in this definition, but more generally, I would suggest proving a TFAE for an additive functor between abelian categories:
> 
> 1. If `S` is _short_ exact, then `S.map F` is exact and `F.map S.f` is mono;
> 2. If `S` is exact and `S.f` is mono, the same applies to `S.map F`.
> 3. `F` preserves kernels.
> 4. `F` preserves finite limits.
> 
> Using the dual result, we should get that `F` preserves `ShortComplex.ShortExact` iff it preserves both finite limits and finite colimits, and with some more work we would also obtain that this is equivalent to preserving `ShortComplex.Exact` (and preserving homology). (You have obtained this, but it is also an opportunity to formalise criteria for left exact and right exact functors.)

This is done now! Thanks for the suggestion

Reply:
Great. But, you need to merge with master and the proofs can be improved...